aws-ssm-parameter/src/main/java/com/amazonaws/ssm/parameter/CreateHandler.java

```diff
@@ -0,0 +1,91 @@
+package com.amazonaws.ssm.parameter;
+
+import org.apache.commons.lang3.RandomStringUtils;
+
+import software.amazon.awssdk.services.ssm.SsmClient;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.delay.Constant;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Random;
+
+public class CreateHandler extends BaseHandlerStd {
+    @Override
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<SsmClient> proxyClient,
+            final Logger logger) {
+        final ResourceModel model = request.getDesiredResourceState();
+
+        Map<String, String> consolidatedTagList = request.getDesiredResourceTags();
+        consolidatedTagList.putAll(request.getSystemTags());
+
+        // Set model primary ID if absent
+        if(model.getName() == null) {
+            model.setName(generateParameterName(
+                    request.getLogicalResourceIdentifier(),
+                    request.getClientRequestToken()
+            ));
+        }
+
+        if(model.getDataType() != null && model.getDataType() == Constants.AWS_EC2_IMAGE_DATATYPE) {
```

Comment:
the only difference is that u set a custom delay strategy in this case? if so can u set it out side of the chain via if statement - default otherwise and then put that in a single chain

Reply:
The difference also includes a stabilization handler for this case, like one case stabilization handler is not required but for the other case it is required, can you please let me know how I can proceed with that?